### DISCUSSION TITLE
SecretManager -> use secrets for Redis elasticache, amazon posgress db, importing many secrets

### DISCUSSION THREAD

**Discussion by Xen85** [awspring/spring-cloud-aws#538] [Q&A]
Hello everyone,
I'm using version 2.4.x
i need to use a secret to configure my redis elasticache and another one to configure my posgress db.

I watched this video https://youtu.be/gIbr6-AR6T8 that it's possibile to use secret to configure posgress.
I also watched this video https://youtu.be/2WwR2zbkQdQ where you described RDS service, is it possible to use secret manager to connect to the RDS ?

I've saw another article https://reflectoring.io/spring-cloud-aws-redis/ to connect to redis for elasticache but I need to do the same via secrets.

Another question about secretsManager, if i have many secretis to import with same property names (as "password" for example), is it possible to import them with a different prefix in my properties?
example : secret1.password, secret2.password?

Thank you for answering!

**Comment by MatejNedic** [comment]
Hey @Xen85,

check out our sample how to call secrets manager [here](https://github.com/awspring/spring-cloud-aws/blob/main/spring-cloud-aws-samples/spring-cloud-aws-secrets-manager-sample/src/main/resources/application.yml).
You will need to use `spring.config.import` in 2.4 and after that, you can reference secrets inside of YAML or properties with `${secret_key_inside_json}`.

Regarding naming secrets with the same name you will have to use JSON structure inside AWS and have keys named differently.
Or just call the secrets by a different name.
For example, let's say you have 2 secrets of the JSON structure. 
The first secret is named: `/config/rds`

```
{
"rds-password": "somePass"
}
```

The second secret is named: `/config/redis`

```
{
"redis-password": "somePass"
}
```

If you would call `spring.config.import=aws-secretsmanager:/config/redis;/config/rds`.

Inside your properties, you can resolve passwords like:
`redis.password=${redis-password}`

Hope this helps. If you have more questions feel free to write. :)

**Reply by Xen85** [reply]
Hello, thanks for answering.
my problem is that i don't have any control about secrets i must read.
So it's impossible to me rule the name of the secrets or the name of json property inside the secret.

However about posgress and redis is there another way to use some solution out of the box like in the video?

**Reply by maciejwalkowiak** [reply]
You can still use https://github.com/aws/aws-secretsmanager-jdbc for postgres. For Redis you would have to implement something custom. We don't offer it out of the box in Spring Cloud AWS.

Regarding colliding names, we don't offer it now but if there's is a well thought through PR for the main branch (3.x) we can include it.

**Reply by Xen85** [reply]
so ther's not any way to control the connection with elasticache via redis ?

That's a real bad news ._.

however thanks for answering!
